brave/sync
# Protobuf.JS 6.45; SyncRecord as JS object

Fix #6 

1. Ensures SyncRecord IPC message compatibility.

2. Upgrades protobuf.js to 6.4.5 and converts api.proto to static JS code.
  - Proto: With classes as static code you can't use reflection e.g. `SyncRecord.lookup('Site')`. This is minor beause you'd just do `SyncRecord.Site`.
  - Proto: `oneof` attributes have issues, notably SyncRecord .objectData is stuck as "bookmark". As a workaround I made a helper: serializer.getSyncRecordObjectData()

Test plan for (1):
- `yarn dist` to browser-laptop `#feature/syncing`
- Go to [browser-laptop app/sync.js#L236](https://github.com/brave/browser-laptop/blob/feature/syncing/app/sync.js#L236); change RECEIVE_SYNC_RECORDS to GET_EXISTING_OBJECTS and add `console.log(records)`.
- Start browser-laptop and visit any URL to make a historySite.
- Look at the console log and compare the record in `sending record` with the GET_EXISTING_OBJECTS record. They should be the same.
